contracts/FIFOMarket.sol

```diff
@@ -347,16 +356,15 @@ contract FIFOMarket is
           i == totalNumberActiveRemovals - 1 &&
           remainingAmountToFill > removalAmount
         ) {
-          revert("Market: Not enough supply");
+          revert InsufficientSupply();
         }
         ids[numberOfRemovals] = removalId;
         amounts[numberOfRemovals] = removalAmount; // this removal is getting used up
         suppliers[numberOfRemovals] = _currentSupplierAddress;
         remainingAmountToFill -= removalAmount;
-        require(
-          _activeSupply[_currentSupplierAddress].removeRemoval(removalId),
-          "Market: Failed to remove removal from supply"
-        ); // TODO (Gas Optimization): Use custom error
+        if (!_activeSupply[_currentSupplierAddress].removeRemoval(removalId)) {
+          revert FailedToRemoveRemovalFromSupply({removalId: removalId});
```

Comment:
https://github.com/nori-dot-eco/contracts/pull/280/files#r913286493

Reply:
isn't this actually redundant given [this block of code](https://github.com/nori-dot-eco/contracts/pull/280/files#diff-15c315f8423ca6a9233aa5946874fc34024b7a4c80f9a6cd5c102bfc3cfa35f8R58-R65)

Reply:
Why does this use a different error than [this line](https://github.com/nori-dot-eco/contracts/pull/280/files#diff-ddd1c3489433b771aafb62a8b29f1334d571292323137759c94ad4e1545dad2bR531)?

Reply:
cleaned all of this up by letting things revert inside `removeRemoval` and in `insertRemovalByVintage` if there's an exceptional case and not up here